assets/src/models/detourShapeData.ts

```diff
@@ -0,0 +1,20 @@
+import { array, Infer, number, type } from "superstruct"
+import { DetourShape } from "../detour"
+
+export const DetourShapeData = type({
+  coordinates: array(
+    type({
+      lat: number(),
+      lon: number(),
+    })
+  ),
+})
+export type DetourShapeData = Infer<typeof DetourShapeData>
+
+export const detourShapeFromData = (
+  shapeData: DetourShapeData
+): DetourShape => ({
+  coordinates: shapeData.coordinates.map((pointData) => {
+    return { lat: pointData.lat, lon: pointData.lon }
```

Comment:
**suggestion:** TypeScript syntax can actually make this even more concise:
```suggestion
    return { ...pointData }
```

Reply:
Oh, if TypeScript likes that then that's good too. If that works you could also get rid of the `{ ... }` curly braces and `return` statement. But I'm not sure if it's quite smart enough for that.

Reply:
Wait, I think at that point, we could just get rid of the `.map` call altogether. 🤯